### DISCUSSION TITLE
single user type - migration?

### DISCUSSION THREAD

**Discussion by andyburgessmd** [Budibase/budibase#13705] [General]
Regarding the new single user data type. Has this been applied automatically to what was the previous user type? 
i.e is the .0 in any reference using the user now been removed and if so does that also apply to where it has been used in automation or JS scripts?

**Comment by mike12345567** [comment]
Hi @andyburgessmd - this is a new user type that we have added, to avoid any breaking changes with the old user type. The old user type has become the `multi user` type - it used to have the ability to select between only holding a single user, or multiple, we've removed the ability to select the single side of it, but any existing columns should be totally unchanged - single ones should continue to work exactly as they had been.

We created the new type as the single user type was causing confusion and we decided going forward we will break the types into multi/single where you could have both, allowing us to set stricter rules about how this works in the various supported DBs as well as being able to improve the binding experience for these (another example is attachments, which received the same treatment).

We always try to avoid migrating/changing users' bindings, as its very difficult for us to know what the intention really was, especially when it comes to JS, it would be very difficult for us to write any kind of accurate migration mechanism which doesn't come with a high level of risk.

**Reply by andyburgessmd** [reply]
In my application in the data builder view I am seeing the old user types I have showing as the single user type.
![image](https://github.com/Budibase/budibase/assets/110813989/9b354d62-0c0c-4bc4-a4b8-f68a7de3e53b)
Based on above these should be `multi user`?

**Reply by mike12345567** [reply]
Ah - that might be a UI/UX issue, over how the frontend interprets old types - I'll check with the team, but we haven't performed any migrations so the data/bindings hasn't changed under the hood!
